### DISCUSSION TITLE
Get store from ApiProvider

### DISCUSSION THREAD

**Discussion by donald-richardson-dxd** [reduxjs/redux-toolkit#3091] [Q&A]
How do I get the store that is created from the ApiProvider?

**Comment by markerikson** [comment]
99% of the time, you _shouldn't_ be using `<ApiProvider>`.  Its only purpose is if you want to use RTK Query in an app that _doesn't_ use Redux for anything else whatsoever.  All `<ApiProvider>` does is take an RTK Query API definition and make a store for you.

Normally, you'd actually create a Redux store, render the React-Redux `<Provider store={store}>` around your app, and go from there.

**Reply by phryneas** [reply]
Or, to be more clear: using `ApiProvider` is great if all you have is a single api.  
But in that case, you should never have a need to interact with `store` directly.

If you want to interact with `store`, it's probably best that you create your own. 

Technically though, you can always use `useStore` in a client component. But it's really not recommended.

**Comment by donald-richardson-dxd** [comment]
Got it. The reason I needed access to the store is because I needed to also access data outside of react, using dispatch. I'll set up a store. Thanks.


________________________________
